### PR TITLE
Refactor tree

### PR DESCRIPTION
**Pull request by pablogs9**
Refactor micro_ros_build folder structure. Please, reviewers, give us your opinion about this restructuring and about these points:

- What should we do with the documents under `doc` folder? Do they need to be updated? -> Decision: Delete
- Removed `.gitlab-ci.yml` in foxy. Our CI is in Github.
- Removed `.gitmodules`  since empty.
- Removed `20-default.list` in since it is not used and its URL are 404 not found.
- Modified `3rd-party-licenses.txt` to remove `nuttx_kconfig_vendor`.
- Merged `README.md` and `micro_ros_setup/README.md`
- Simplification of `README.md` support tables
- Removed `client-colcon.metas` because each platform has its own.
- Added `CODEOWNERS` to keep track of maintainer of each platform support. Please add yourself to where you consider.
- Should we add some other maintainer in addition of Ingo@Bosch? Should we modify this?

### CODE REVIEW

**Comment by pablogs9** [comment]
Please @ralph-lange give us feedback about this PR that simplifies the folder structure of the build system package.

**Comment by pablogs9** [comment]
I completely agree with the naming issues. I'm thinking about the places where we have linked `https://github.com/micro-ROS/micro-ros-build` and we should change it to `https://github.com/micro-ROS/micro_ros_setup`.

What about changing the package name to `micro-ros-build`? 

Personally I prefer the first option, I should check some things before changing the micro-ROS entrypoint URL...

What do you think @jamoralp @FranFin?

**Comment by pablogs9** [comment]
It seems that there is no problem with repo renaming: https://github.blog/2013-05-16-repository-redirects-are-here/

**Comment by ralph-lange** [comment]
Renaming the package to micro-ros-build is not an option since dashes are not allowed in package names - although I would prefer the name "build" over "setup". Let's rename the repo to micro_ros_setup.

**Comment by pablogs9** [comment]
Ok, let's merge this PR on Monday and I will change the repo name and check everything

File: README.md [hunk 1]
```diff
@@ -1,29 +1,190 @@
...
+Please note that NuttX with Olimex STM32-E407 board is the reference platform and not everything might be supported on other platforms.
 
-See [micro_ros_setup README](micro_ros_setup/README.md) for usage instructions.
+*\* Support for compiling apps in a native Linux host for testing and debugging*
 
-## Purpose of the project
+# Dependencies
 
-The software is not ready for production use. It has neither been developed nor tested for a specific use case. However, the license conditions of the applicable Open Source licenses allow you to adapt the software to your needs. Before using it in a safety relevant setting, make sure that the software fulfills your requirements and adjust it according to any applicable safety standards (e.g. ISO 26262).
+This package targets **ROS 2** installation. ROS 2 supported distributions are:
```
Comment (FranFin) [inline README.md:37]:
```suggestion
This package targets the **ROS 2** installation. ROS 2 supported distributions are:
```

File: README.md [hunk 2]
```diff
@@ -1,29 +1,190 @@
...
+rosdep update && rosdep install --from-path src --ignore-src -y
+
+colcon build
+
+source install/local_setup.bash
+```
+
+Once the package is built, the firmware scripts are ready to run.
+
+You can find first steps tutorials in [micro-ROS webpage](https://micro-ros.github.io/docs/tutorials/core/first_application_rtos/).
```
Comment (FranFin) [inline README.md:71]:
```suggestion
You can find tutorials for moving your first steps with micro-ROS on an RTOS in the [micro-ROS webpage](https://micro-ros.github.io/docs/tutorials/core/first_application_rtos/).
```

File: README.md [hunk 3]
```diff
@@ -1,29 +1,190 @@
...
+
+
+## Creating micro-ROS firmware
+
+Using `create_firmware_ws.sh [RTOS] [Platform]` command a firmware folder will be created with the required code for building a micro-ROS app. For example, for our reference platform, the invocation is:
```
Comment (FranFin) [inline README.md:76]:
```suggestion
Using the `create_firmware_ws.sh [RTOS] [Platform]` command, a firmware folder will be created with the required code for building a micro-ROS app. For example, for our reference platform, the invocation is:
```

File: README.md [hunk 4]
```diff
@@ -1,29 +1,190 @@
...
+# Creating a FreeRTOS + micro-ROS firmware workspace
+ros2 run micro_ros_setup create_firmware_ws.sh freertos olimex-stm32-e407
+
+# Creating a Zephyr + micro-ROS firmware workspace
+ros2 run micro_ros_setup create_firmware_ws.sh zephyr olimex-stm32-e407
+```
+
+## Configuring micro-ROS firmware
+
+By running `configure_firmware.sh` command the installed firmware is configured and modified in a pre-build step. Usually this command will show its usage if parameters are required:
```
Comment (FranFin) [inline README.md]:
```suggestion
By running the `configure_firmware.sh` command, the installed firmware is configured and modified in a pre-build step. Usually this command will show its usage if parameters are required:
```

Comment (FranFin) [reply README.md]: It's not really clear to me what the sentence "Usually this command will show its usage if parameters are required:" means

File: README.md [hunk 5]
```diff
@@ -1,29 +1,190 @@
...
+By running this command without any argument the available demo applications and configurations will be shown.
+
+Common options available at this configuration step are:
+  - `--transport` or `-t`: `udp`, `serial` or any hardware specific transport label
+  - `--dev` or `-d`: agent string descriptor in a serial-like transport
+  - `--ip` or `-i`: agent IP in a network-like transport
+  - `--port` or `-p`: agent port in a network-like transport
+
+
+Please note that each RTOS has its configuration approach that you might use for further customization of these base configurations. Visit [micro-ROS webpage](https://micro-ros.github.io/docs/tutorials/core/first_application_rtos/) for a detailed information about RTOS configuration.
```
Comment (FranFin) [inline README.md:106]:
```suggestion
Please note that each RTOS has its configuration approach that you might use for further customization of these base configurations. Visit the [micro-ROS webpage](https://micro-ros.github.io/docs/tutorials/core/first_application_rtos/) for detailed information about RTOS configuration.
```

File: README.md [hunk 6]
```diff
@@ -1,29 +1,190 @@
...
+By running `build_firmware.sh` the firmware is built:
+
+```
+ros2 run micro_ros_setup build_firmware.sh
+```
+
+## Flashing micro-ROS firmware
+
+In order to flash the target platform run `flash_firmware.sh` command.
+This step may need some platform-specific procedure to boot the platform in flashing node:
```
Comment (FranFin) [inline README.md:131]: flashing node or flashing mode?

File: README.md [hunk 7]
```diff
@@ -1,29 +1,190 @@
...
+
+As it is explained along this document, the firmware building system takes **four steps**: creating, configuring, building and flashing.
+
+New combinations of platforms and RTOS are intended to be included in `config` folder. For example, the scripts for building a **micro-ROS** app for **Crazyflie 2.1** using **FreeRTOS** is located in `config/freertos/crazyflie21`.
+
+This folder contains up to four scripts:
+- `create.sh`: gets a variable named `$FW_TARGETDIR` and installs in this path all the dependencies and code required for the firmware.
+- `configure.sh`: modifies and configure parameters of the installed dependencies. This step is **optional**.
+- `build.sh`: builds the firmware and create a platform-specific linked binary.
+- `flash.sh`: flash the binary in the target platform.
```
Comment (FranFin) [inline README.md:158]:
```suggestion
- `flash.sh`: flashes the binary in the target platform.
```

File: README.md (hunk 6 above)
Comment (pablogs9) [reply README.md:131]:
```suggestion
This step may need some platform-specific procedure to boot the platform in flashing mode:
```

File: README.md (hunk 4 above)
Comment (pablogs9) [reply README.md]: Modified

**Comment by pablogs9** [comment]
friendly ping @ralph-lange :)

**Pull request merged**